src/bin/close_preopen.rs

```diff
@@ -1,59 +1,60 @@
 use libc;
 use misc_tests::open_scratch_directory;
-use misc_tests::wasi::{wasi_fd_close, wasi_fd_fdstat_get, wasi_fd_renumber};
```

Comment:
Is `wasi_fd_fdstat_get` staying in this crate simply because we've not added it to the `wasi` crate yet?

Reply:
It's in the `wasi` crate already, but there it returns the `fdstat_t` as a return value rather than as a `&mut` output parameter. We have tests which depend on being able to pass in a pointer to existing memory so that they can test that the WASI implementation does actually write to the memory. In these cases, I've left the wrappers in wasi_misc_tests's wasi_wrappers.rs file.

In the future, I expect WASI itself will switch to multiple return values (once wasm has them), and this whole buisness of out parameters can go away.